Firmware/mmu2.cpp

```diff
@@ -676,25 +681,26 @@ void MMU2::OnMMUProgressMsg(ProgressCode pc){
                 switch ( WhereIsFilament() )
                 {
                 case FilamentState::AT_FSENSOR:
-                    // fsensor triggered, stop moving the extruder
+                    // fsensor triggered, finish FeedingToBondtech state
                     loadFilamentStarted = false;
-                    // TODO: continue to ProgressCode::FeedingToNozzle?
+                    // After the MMU knows the FSENSOR is triggered it will:
+                    // 1. Push the filament by additional 30mm (see fsensorToNozzle)
+                    // 2. Disengage the idler and push another 5mm.
+                    current_position[E_AXIS] += 30.0f + 5.0f;
+                    plan_buffer_line_curposXYZE(MMU2_LOAD_TO_NOZZLE_FEED_RATE);
                     break;
                 case FilamentState::NOT_PRESENT:
                     // fsensor not triggered, continue moving extruder
-                    // TODO: Verify what's the best speed here?
-                    current_position[E_AXIS] += MMU2_LOAD_TO_NOZZLE_FEED_RATE;
+                    current_position[E_AXIS] += 5.0f;
```

Comment:
This looks like speculative and dependent on the frequency of Queries being sent over the communication protocol. I.e. if we ever change the frequency of requests, the 5mm may not be true anymore.

We may need a better solution in the future.